### PR TITLE
Update rust and add update script

### PR DESCRIPTION
**Pull request by bradjc**
This bumps the rust version to a nightly that actually includes clippy. It also adds a script to make this updating process easier in the future.

This separates this from the other clippy stuff since netlify is failing since the old rust version doesn't include clippy.

### Testing Strategy

This pull request was tested by well nothing really.

### TODO or Help Wanted

Nope

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

File: tools/update_rust_version.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# Ask rustup to pick the latest version that will work.
+# This requires rustup >= 1.20.0.
+echo "Updating rustc to latest compatible version..."
+rustup update nightly >/dev/null 2>/dev/null
+
+# Rerun the command so that it prints out the version it installed. We then have
+# to extract that from the output. If there is a better way to do this then we
+# should update this.
+RUSTUP_NIGHTLY_VERSION=`rustup update nightly 2>/dev/null`
+BEST_DATE=`echo $RUSTUP_NIGHTLY_VERSION | sed 's/.* \([0-9]*-[0-9]*-[0-9]*\).*/\1/g'`
```
Comment (gendx) [inline tools/update_rust_version.sh]: How is this the "best" date, as opposed to the latest date? Does `rustup update nightly` automatically picks a version compatible with all installed tools (rustfmt, clippy)? Or does it happily break them (since nightly is unstable anyway)?

Comment (bradjc) [reply tools/update_rust_version.sh]: > Does rustup update nightly automatically picks a version compatible with all installed tools (rustfmt, clippy)?

Yep!

File: chips/cc26x2/src/ccfg.rs [hunk 2]
```diff
@@ -99,7 +99,7 @@ impl Registers {
 }
 
 register_bitfields![
-    u8,
+    u32,
```
Comment (gendx) [inline chips/cc26x2/src/ccfg.rs]: How is this change related to updating rust nightly?

Comment (gendx) [reply chips/cc26x2/src/ccfg.rs]: So the previous type was incorrect, right? As I understand it, there are offsets greater than 8 so the associated fields wouldn't be representable in a `u8`, right?

If that's the case, is there a way/plan to add unit tests to the `register_bitfields` to check that e.g. offsets fit in the underlying integer type, that they don't overlap with one another, etc?

Comment (bradjc) [reply chips/cc26x2/src/ccfg.rs]: Right, as far as I can tell `u8` is wrong. As for additional tests/checks...I'm not sure.
